### PR TITLE
onDisconnect listener + off() EventManager unsub

### PR DESCRIPTION
**Pull request by W3stside**
1. Adds `onDisconnect` method / listener
2. Adds `off()` EventManager and Core method for unsubscribing

To test: load example, connect with mobile test wallet, then disconnect. It should reset state and show user front page. Attempting to connect again prompts QR code

### CODE REVIEW

**Comment by W3stside** [comment]
shout out to @velenir for helping co-author some of this :+1:

File: src/core/index.tsx [hunk 1]
```diff
@@ -31,15 +31,27 @@ class Core {
...
   }
 
-  public on(event: string, callback: (result: any) => void): void {
+  public on(event: string, callback: (result: any) => void): () => void {
     this.eventManager.on({
       event,
       callback
     });
+
+    return () => this.eventManager.off({
```
Comment (pedrouid) [inline src/core/index.tsx]: Why do you return a callback to remove it? Just curious

**Comment by pedrouid** [comment]
Take a look too when you can cc @crisgarner

File: src/core/index.tsx (hunk 1 above)
Comment (Velenir) [reply src/core/index.tsx]: I think it's a good practice to make `unsubscribe` function readily available. For example, for easier cleanup in a `useEffect` hook:
```
const core = new Web3Connect.Core({
...
useEffect(() => {
  return core.on('connect', provider => setWeb3(new Web3(provider)))
}, [])
```
No need to keep a reference to ` provider => setWeb3(new Web3(provider))` to pass to `core.off('connect',...)`

**Comment by pedrouid** [comment]
Hey @W3stside, thanks again for the PR. Sorry this has been hanging for so long because I had to prioritize other more critical refactors that were on-going. Just resolved some merge conflicts and going to take a look at this today to include on the next release!

**Comment by lukasschor** [comment]
> Hey @W3stside, thanks again for the PR. Sorry this has been hanging for so long because I had to prioritize other more critical refactors that were on-going. Just resolved some merge conflicts and going to take a look at this today to include on the next release!

Is this still scheduled for the next release? It would be very helpful to have this merged. 👍

**Comment by lukasschor** [comment]
Any update on this @crisgarner?

**Comment by crisgarner** [comment]
Merged now! will be published on next release

**Comment by lukasschor** [comment]
@crisgarner When is the next release expected?

**Comment by crisgarner** [comment]
> @crisgarner When is the next release expected?

Working on it right now, it should be out in a couple of minutes

**Comment by crisgarner** [comment]
@lukasschor It is released! Thanks!

**Pull request merged**